frontend/components/forms/admin/AppConfigForm/AppConfigFormFunctional.tsx

```diff
@@ -0,0 +1,1121 @@
+    smtpAuthenticationMethod:
+      formData.smtp_settings.authentication_method || "",
+    // Global agent options
+    agentOptions: yaml.dump(formData.agent_options) || {},
+    // Host status webhook
+    enableHostStatusWebhook:
+      formData.webhook_settings.host_status_webhook
+        .enable_host_status_webhook || false,
+    hostStatusWebhookDestinationURL:
+      formData.webhook_settings.host_status_webhook.destination_url || "",
+    hostStatusWebhookHostPercentage:
+      formData.webhook_settings.host_status_webhook.host_percentage ||
+      undefined,
+    hostStatusWebhookDaysCount:
+      formData.webhook_settings.host_status_webhook.days_count || undefined,
+    // Usage statistics
+    enableUsageStatistics: formData.server_settings.enable_analytics,
+    // Advanced options
+    domain: formData.smtp_settings.domain || "",
+    verifySSLCerts: formData.smtp_settings.verify_ssl_certs || false,
+    enableStartTLS: formData.smtp_settings.enable_start_tls,
+    enableHostExpiry:
+      formData.host_expiry_settings.host_expiry_enabled || false,
+    hostExpiryWindow: formData.host_expiry_settings.host_expiry_window || 0,
+    disableLiveQuery: formData.server_settings.live_query_disabled || false,
+  });
+
+  const {
+    orgName,
+    orgLogoURL,
+    serverURL,
+    enableSSO,
+    idpName,
+    entityID,
+    issuerURI,
+    idpImageURL,
+    metadata,
+    metadataURL,
+    enableSSOIDPLogin,
+    enableSMTP,
+    smtpSenderAddress,
+    smtpServer,
+    smtpPort,
+    smtpEnableSSLTLS,
+    smtpAuthenticationType,
+    smtpUsername,
+    smtpPassword,
+    smtpAuthenticationMethod,
+    agentOptions,
+    enableHostStatusWebhook,
+    hostStatusWebhookDestinationURL,
+    hostStatusWebhookHostPercentage,
+    hostStatusWebhookDaysCount,
+    enableUsageStatistics,
+    domain,
+    verifySSLCerts,
+    enableStartTLS,
+    enableHostExpiry,
+    hostExpiryWindow,
+    disableLiveQuery,
+  } = iterateFormData;
+
+  // OLD
+  const [formErrors, setFormErrors] = useState<IAppConfigFormErrors>({});
+
+  // █▀▀ █▀█ █▀█ █▀▄▀█   █▀▀ █░█ ▄▀█ █▄░█ █▀▀ █▀▀
+  // █▀░ █▄█ █▀▄ █░▀░█   █▄▄ █▀█ █▀█ █░▀█ █▄█ ██▄
+  const handleInputChange = ({ name, value }: IFormField) => {
+    setIterateFormData({ ...iterateFormData, [name]: value });
+  };
+
+  console.log("iterateFormData", iterateFormData);
+
+  // ▀█▀ █▀█ █▀▀ █▀▀ █░░ █▀▀   █▀▄▀█ █▀█ █▀▄ ▄▀█ █░░ █▀
+  // ░█░ █▄█ █▄█ █▄█ █▄▄ ██▄   █░▀░█ █▄█ █▄▀ █▀█ █▄▄ ▄█
+
+  const toggleHostStatusWebhookPreviewModal = () => {
+    setShowHostStatusWebhookPreviewModal(!showHostStatusWebhookPreviewModal);
+    return false;
+  };
+
+  const toggleUsageStatsPreviewModal = () => {
+    setShowUsageStatsPreviewModal(!showUsageStatsPreviewModal);
+    return false;
+  };
+
+  // █▀▀ █▀█ █▀█ █▀▄▀█   █▀ █░█ █▄▄ █▀▄▀█ █ ▀█▀
+  // █▀░ █▄█ █▀▄ █░▀░█   ▄█ █▄█ █▄█ █░▀░█ █ ░█░
+  const onFormSubmit = () => {
+    // Validators
+    const errors: any = {};
+
+    if (enableSSO) {
```

Comment:
`useReducer` is the newer pattern for form validation. But, that pattern is only implemented in cases where we have a context object (app, query, policy). I'm looking into how we want to implement it here, but I'll create a separate ticket for that. For our purposes, this validation works and is totally fine for this ticket.

Reply:
I noticed this same code in `validate.js` that the old `AppConfigForm` was using. Can we keep using the imported file for validation to cut down on line size in this component?

Reply:
As I mentioned in my video, we'd have to pass formData into validate everywhere and then destructure on a separate file which ends up adding a lot of extra lines of code everywhere. I cleaned this file down to 1078 lines from 1300 which is great!